marketdata/stream/client.go

```diff
@@ -381,7 +381,8 @@ func (c *client) maintainConnection(ctx context.Context, u url.URL, initialResul
 // isErrorIrrecoverable returns whether the error is irrecoverable and further retries should
 // not take place
 func isErrorIrrecoverable(err error) bool {
-	return errors.Is(err, ErrInvalidCredentials) || errors.Is(err, ErrInsufficientSubscription)
+	return errors.Is(err, ErrInvalidCredentials) || errors.Is(err, ErrInsufficientSubscription) ||
```

Comment:
[minor] At this point I'd opt for creating a set of irrecoverable errors and iterate over them here.

Reply:
Good idea for the future 😉